src/stac_utils/bsd.py

```diff
@@ -0,0 +1,150 @@
+import hmac, hashlib
+import os, time
+import sys
+
+import xmltodict
+import requests
+import pandas as pd
+
+from insert_leads import insert_hot_lead
+from datetime import datetime, date, timedelta
+
+if "BSD_API_SECRET" not in os.environ:
+    # Try to load from .env
+    from dotenv import load_dotenv
+
+    load_dotenv()
+
+try:
+    api_secret = os.environ['BSD_API_SECRET']
+    BSD_API_ID = os.environ['BSD_API_ID']
+    BSD_URL = os.environ['BSD_URL']
+except KeyError:
+    sys.exit('Cannot find BSD API Secret')
```

Comment:
this file should all be converted to a subclass of HTTPClient with __init__ and create_session, maybe generate_api_mac as a thing called by create_session, and more stuff from below

should also get test coverage, docstrings, and added to docs/index.rst